30412/Ilie Oana - Andreea/assignment1/main.c

```diff
@@ -0,0 +1,72 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "sll.h"
+
+#define FILE_PATH_INPUT "../input.dat"
+#define FILE_PATH_OUTPUT "../output.dat"
+#define MAX_LINE 20
+
+int main() {
+    FILE *inputFile;
+    inputFile = fopen(FILE_PATH_INPUT, "r");
+    if (inputFile == NULL) {
+        perror("Error opening file");
+        return (-1);
+    }
+
+    FILE *outputFile;
+    outputFile = fopen(FILE_PATH_OUTPUT, "w");
+
+    initializeSLL();
+
+    char *line, *p, *operation;
+    int x;
+    line = (char *) malloc(MAX_LINE * sizeof(char));
+    operation = (char *) malloc(MAX_LINE * sizeof(char));
+    while (!feof(inputFile)) {
+        fgets(line, MAX_LINE, inputFile);
+        p = strtok(line, " ");
+        strcpy(operation, p);
+        p = strtok(0, " ");
+        if (p)
+            x = atoi(p);
+        if (strcmp(operation, "AF") == 0)
+            AF(x);
+        else {
```

Comment:
Uh, there's a much cleaner way to write this. Instead of
```
else {
    if ( ... )
}
...
```
You can just use
``` 
else if (...) { ... }
``` 
Then you would have this in one line and not increase indentation like this